File: src/3-Problem11.py
"""
    NIM/Nama  : 13516002/Antonio Setya
    Topik     : Prosedur/Fungsi
    Deskripsi : Menentukan apakah angka masukan merupakan bilangan menanjak atau bukan
"""

# Fungsi isMenanjak :
# I.S. Menerima string yang berisi bilangan masukan
# F.S. Mengembalikan true jika bilangan menanjak, false jika tidak
def isMenanjak(inp):
    i = len(inp) - 1
    result = True
    while ((i > 0) and result):
        if (int(inp[i]) < int(inp[i-1])):
            i -= 1
        else:
            result = False
    return result

# Input
n = input("Masukkan N: ") # Menyimpan bilangan N.

# Memroses N dan output
if (isMenanjak(n)):
    print("Bilangan " + n + " adalah bilangan menanjak.")
else:
    print("Bilangan " + n + " bukan bilangan menanjak.")

File: src/3-Problem11.cpp
#include <iostream>

using namespace std;

bool IsMenanjak(int n, int m){
		if ((n/10) ==0) {
			return true;
		}else{
			if ((n/m) < 10){
				if (n%10 < ((n/10)%10)){
					return true;
				}else{
					return false;
				}
			}else{
				
				if (((n/m)%10)<((n/(m*10))%10)){
					return IsMenanjak(n, m*10);
				}else{
					return false;
				}
			}	
		}
	}


int main(){

	int n;

	cout<<"Masukan N: ";
	cin>> n;

	cout<<"Bilangan "<<n;
	if (IsMenanjak(n,10)){
		cout<<" adalah bilangan menanjak."<<endl;
	}else{
		cout<<" bukan bilangan menanjak."<<endl;
	}
	
	return 0;
}
